src/rgw/rgw_datalog.h

```diff
@@ -234,14 +234,16 @@ class RGWDataChangesLog {
 
   using ChangeStatusPtr = std::shared_ptr<ChangeStatus>;
 
-  lru_map<rgw_bucket_shard, ChangeStatusPtr> changes;
+  lru_map<std::pair<rgw_bucket_shard, uint64_t>, ChangeStatusPtr> changes;
```

Comment:
nit: this pair is worth a typedef, since it's also used below for `cur_cycle`, and for a local variable in `RGWDataChangesLog::renew_entries()`

Reply:
FIXED.